phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -134,6 +137,7 @@ public TableResultIterator(MutationState mutationState, Scan scan, ScanMetricsHo
         this.retry=plan.getContext().getConnection().getQueryServices().getProps()
                 .getInt(QueryConstants.HASH_JOIN_CACHE_RETRIES, QueryConstants.DEFAULT_HASH_JOIN_CACHE_RETRIES);
         IndexUtil.setScanAttributesForIndexReadRepair(scan, table, plan.getContext().getConnection());
+        scan.setAttribute(BaseScannerRegionObserver.SERVER_PAGING, TRUE_BYTES);
```

Comment:
Can there be cases where SERVER_PAGING is not desirable? This change sort of enforces all queries to use paging. Wondering if it is worth introducing a client-side config to decide whether to set this attribute to True/False.

Reply:
I added so that we do not need to upgrade the clients for this change and it can work with mix of old and new clients. Do you have another suggestion here? To disable paging, we can set the page size to Long.MAX_VALUE

Reply:
Makes sense. For old clients this attribute won't be set so it will be like current behavior. For new clients we can set page size to Long.MAX_VALUE to disable it. 👍